src/format/OPUXReader.cpp

```diff
@@ -126,9 +126,13 @@ namespace
                 const auto valueMap = fieldMap.value("value").toMap();
                 const auto key = valueMap.firstKey();
                 if (key == "totp") {
-                    // Build otpauth url
-                    QUrl otpurl(QString("otpauth://totp/%1:%2?secret=%3")
-                                    .arg(entry->title(), entry->username(), valueMap.value(key).toString()));
+                    auto totp = valueMap.value(key).toString();
+                    if (!totp.startsWith("otpauth://")) {
+                        // Build otpauth url
+                        QUrl url(
+                            QString("otpauth://totp/%1:%2?secret=%3").arg(entry->title(), entry->username(), totp));
```

Comment:
I think we might have to url encode the parameters just to be safe.

Reply:
all fixed (both 1PUX and Bitwarden)